pkg/jsontabwriter/jsontabwriter.go

```diff
@@ -0,0 +1,204 @@
+package jsontabwriter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/json2table"
+	"github.com/spf13/viper"
+)
+
+const (
+	outputFormatErr = "invalid format: %s"
+
+	// JSONFormat defines the JSON format.
+	JSONFormat = "json"
+	// TextFormat defines a human-readable format.
+	TextFormat = "text"
+)
+
+// TODO: remove cols as function parameter once --cols flag fix is ready
+func GenerateOutput(rootPath string, jsonPaths map[string]string, obj interface{}, cols []string) (string, error) {
+	if viper.IsSet(constants.ArgQuiet) {
+		return "", nil
+	}
+
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == JSONFormat {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == TextFormat {
+		return generateTextOutputFromJSON(rootPath, obj, jsonPaths, cols)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateOutputPreconverted(obj interface{}, convertedObj []map[string]interface{}, cols []string) (string, error) {
+	if viper.IsSet(constants.ArgQuiet) {
+		return "", nil
+	}
+
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutputFromTable(convertedObj, cols)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateVerboseOutput(format string, a ...interface{}) string {
+	if viper.IsSet(constants.ArgQuiet) {
+		return ""
+	}
+
+	if !viper.IsSet(constants.ArgVerbose) {
+		return ""
+	}
+
+	msg := fmt.Sprintf("[INFO] "+format, a...)
+
+	return GenerateRawOutput(msg)
+}
+
+func GenerateLogOutput(format string, a ...interface{}) string {
+	msg := fmt.Sprintf(format, a...)
+
+	return GenerateRawOutput(msg)
+}
+
+func GenerateRawOutput(a interface{}) string {
+	if viper.IsSet(constants.ArgQuiet) {
+		return ""
+	}
+
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		out, err := generateJSONOutput(a)
+		if err != nil {
+			return ""
+		}
+
+		return out
+	}
+
+	if outputFormat == "text" {
+		return fmt.Sprintf("%v\n", a)
+	}
+
+	return ""
+}
+
+func generateJSONOutput(obj interface{}) (string, error) {
+	out, err := json.MarshalIndent(obj, "", "\t")
+	if err != nil {
+		return "", err
+	}
+
+	return string(out) + "\n", nil
+}
+
+func generateTextOutputFromJSON(rootPath string, obj interface{}, jsonPaths map[string]string, cols []string) (string, error) {
+	table, err := json2table.ConvertJSONToTable(rootPath, jsonPaths, obj)
+	if err != nil {
+		return "", fmt.Errorf("failed converting object to table using %+v: %w", jsonPaths, err)
+	}
+
+	return writeTableToText(table, cols), nil
+}
+
+func generateTextOutputFromTable(table []map[string]interface{}, cols []string) (string, error) {
```

Comment:
it looks like this func never returns an error. Why not use `writeTableToText` directly though?

Reply:
tbh, I don't remember, but I will change it :))))